mongo/filterstore.go

```diff
@@ -170,22 +171,105 @@ func (s *FilterStore) RemoveFilterDimension(filterID string, name string) error
 }
 
 // AddFilterDimensionOption to a filter
-func (s *FilterStore) AddFilterDimensionOption(*models.AddDimensionOption) error {
+func (s *FilterStore) AddFilterDimensionOption(newOption *models.AddDimensionOption) error {
+	err := s.checkFilterState(newOption.FilterID)
+	if err != nil {
+		return err
+	}
+
+	session := s.Session.Copy()
+	queryOptions := bson.M{"filter_job_id": newOption.FilterID, "dimensions": bson.M{"$elemMatch": bson.M{"name": newOption.Name}}}
+	update := bson.M{"$push": bson.M{"dimensions.0.options": newOption.Option}}
+	err = session.DB(Database).C(FiltersCollection).Update(queryOptions, update)
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return NotFound
+		}
+		return err
+	}
 	return nil
 }
 
 // GetFilterDimensionOptions return a list of dimension options
 func (s *FilterStore) GetFilterDimensionOptions(filterID string, name string) ([]models.DimensionOption, error) {
-	return nil, nil
+	session := s.Session.Copy()
+	queryFilter := bson.M{"filter_job_id": filterID}
+	queryDimension := bson.M{"filter_job_id": filterID, "dimensions": bson.M{"$elemMatch": bson.M{"name": name}}}
+	dimensionSelect := bson.M{"dimensions": 1}
+	var result models.Filter
+	err := session.DB(Database).C(FiltersCollection).Find(queryFilter).Select(dimensionSelect).One(&result)
+	if err != nil {
+		if err == mgo.ErrNotFound {
+			return nil, NotFound
+		}
+		return nil, err
+	}
+	err = session.DB(Database).C(FiltersCollection).Find(queryDimension).Select(dimensionSelect).One(&result)
```

Comment:
Do we need to make another request to mongo? We have the result of the filter job which contains the dimensions and options?

Instead we can check for the dimension by ranging over array of dimensions, this would be faster than making a call to the database again with a different query as well as opening us up to a change to the document between steps (even if it is highly unlikely).